warp

```diff
@@ -56,7 +60,8 @@ warp() {
 
 # allow warp to be sourced without running
 if [[ $_ == $0 ]]; then
-  warp
+  # pass all arguments to the warp function
+  warp $@
```

Comment:
You need double-quotes around $@ to propagate double-quotes to the subprocess.

Not necessary for this, but a good practice.